### PR TITLE
[roseus_smach] rename smach-exec-with-spin -> exec-state-machine

### PR DESCRIPTION
**Pull request by furushchev**
- `smach-exec` and `exec-smach-with-spin` are almost the same functions.
- This pull request merges them into `exec-state-machine`

### Note

- I searched usage of old functions on `jsk-ros-pkg` / `euslib` and it's ok to even remove old functions.

- Old functions are redirected to new function with deprecated warning, and they will be removed in a month.

```lisp
(defun smach-exec (sm)
  "Deprecated function"
  (ros::ros-warn "This function is deprecated and will be removed. Use exec-state-machine")
  (exec-state-machine sm))
(defun exec-smach-with-spin (sm &optional mydata &key (hz 1) iterate)
  "Deprecated function"
  (ros::ros-warn "This function is deprecated and will be removed. Use exec-state-machine")
  (exec-state-machine sm mydata :hz hz :iterate iterate))
```

- Updated readme / sample codes / tests
- add `task_compiler` (heavily depends on `roseus_smach`) to travis test

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,9 +20,11 @@ env:
     - ROS_DISTRO=hydro  USE_DEB=true  ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
     - ROS_DISTRO=hydro  USE_DEB=false NOT_TEST_INSTALL=true
     - ROS_DISTRO=hydro  USE_DEB=true  NOT_TEST_INSTALL=true INSTALL_SRC="http://github.com/jsk-ros-pkg/jsk_pr2eus" TEST_PKGS="pr2eus"
+    - ROS_DISTRO=hydro  USE_DEB=true  NOT_TEST_INSTALL=true INSTALL_SRC="http://github.com/jsk-ros-pkg/jsk_planner" TEST_PKGS="task_compiler"
```
Comment (k-okada) [inline .travis.yml]: @furushchev jsk_planner -> jsk_planning ?

Comment (furushchev) [inline .travis.yml]: @k-okada Oops.. Nice catch, thank you very much!

**Comment by furushchev** [comment]
```
[task_compiler] /opt/ros/hydro/share/downward/scripts/search/unitcost: 4: /opt/ros/hydro/share/downward/scripts/search/unitcost: gawk: not found
```

I may only need to restart test thanks to jsk-ros-pkg/jsk_3rdparty#85 ?

**Comment by k-okada** [comment]
@furushchev Could not pass the test -> https://travis-ci.org/jsk-ros-pkg/jsk_roseus/jobs/202516461

**Comment by furushchev** [comment]
MEMO: waiting for https://github.com/jsk-ros-pkg/jsk_planning/pull/56

**Pull request merged**